source/date_time.c

```diff
@@ -170,21 +171,26 @@ struct tm s_get_time_struct(struct aws_date_time *dt, bool local_time) {
 }
 
 void aws_date_time_init_now(struct aws_date_time *dt) {
-    uint64_t current_time = 0;
-    aws_sys_clock_get_ticks(&current_time);
-    dt->timestamp = (time_t)aws_timestamp_convert(current_time, AWS_TIMESTAMP_NANOS, AWS_TIMESTAMP_SECS, NULL);
+    uint64_t current_time_ns = 0, current_time_ms = 0;
+    aws_sys_clock_get_ticks(&current_time_ns);
+    current_time_ms = aws_timestamp_convert(current_time_ns, AWS_TIMESTAMP_NANOS, AWS_TIMESTAMP_MILLIS, NULL);
+    dt->timestamp = (time_t)(current_time_ms / AWS_TIMESTAMP_MILLIS);
+    dt->milliseconds = (uint16_t)(current_time_ms % AWS_TIMESTAMP_MILLIS);
     dt->gmt_time = s_get_time_struct(dt, false);
     dt->local_time = s_get_time_struct(dt, true);
 }
 
 void aws_date_time_init_epoch_millis(struct aws_date_time *dt, uint64_t ms_since_epoch) {
     dt->timestamp = (time_t)(ms_since_epoch / AWS_TIMESTAMP_MILLIS);
+    dt->milliseconds = (uint16_t)(ms_since_epoch % AWS_TIMESTAMP_MILLIS);
```

Comment:
Same here, should be able to collapse to:

`dt->timestamp = aws_timestamp_convert(ms_since_epoch, AWS_TIMESTAMP_MILLIS, AWS_TIMESTAMP_SECS, &dt->milliseconds);`